lib/util/Components.js

```diff
@@ -486,6 +492,23 @@ function componentRule(rule, context) {
       return null;
     },
 
+    /**
+     * Get a parent scope created by a FunctionExpression or FunctionDeclaration
+     * @param {Scope} scope The child scope
+     * @returns {Scope} A parent function scope
+     */
+    getParentFunctionScope(scope) {
+      scope = scope.upper;
+      while (scope) {
+        const type = scope.block.type;
```

Comment:
will scope always have an object `block` property?

Reply:
I think yes, here's the [proof](https://github.com/eslint/eslint-scope/blob/3e69fdcf102082f3b3c419d0864d7ed90323f4ca/lib/scope.js#L650). There's also no extra checks in [the code above](https://github.com/yannickcr/eslint-plugin-react/pull/1989/files#diff-bafb3355a409c4ba076a9e3d609ad65bR470).